src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -205,8 +227,14 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
          -> InterpResult<'tcx> {
             let global = this.machine.borrow_tracker.as_ref().unwrap().borrow();
             let ty = place.layout.ty;
+            let ty_is_freeze = ty.is_freeze(*this.tcx, this.typing_env());
```

Comment:
`is_freeze` can be expensive, so only execute it when needed. Here, you are only using the result inside the `if`.